test/e2e/utils/rule_builder.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2025 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+)
+
+type RuleBuilder interface {
+	GetIngress() crdv1beta1.Rule
+	GetEgress() crdv1beta1.Rule
+}
+
+type BaseRuleBuilder struct {
+	Protoc               AntreaPolicyProtocol
+	Port                 *int32
+	PortName             *string
+	EndPort              *int32
+	IcmpType             *int32
+	IcmpCode             *int32
+	IgmpType             *int32
```

Comment:
We follow the [Go code convention](https://go.dev/wiki/CodeReviewComments#initialisms) that words in names that are initialisms have a consistent case.
> For example, “URL” should appear as “URL” or “url” (as in “urlPony”, or “URLPony”), never as “Url”. As an example: ServeHTTP not ServeHttp. For identifiers with multiple initialized “words”, use for example “xmlHTTPRequest” or “XMLHTTPRequest”.